### ISSUE TITLE
Target-typed conditional versus nullable reference types

### ISSUE THREAD

**Issue by gafter** [dotnet/roslyn#46063]
The following program should cause a nullable warning, but it doesn't.

```csharp
#nullable enable
class Program
{
    static void Main()
    {
        bool b = false;
        object o = b ? null : 2; // should be warning
    }
}
```

**Comment by RikkiGibson** [comment]
Recent bits (ed2982191a7db2358ae97231b995e792b0eb9a6a) do not repro this issue. I suspect it was fixed by #47405

**Comment by RikkiGibson** [comment]
The squiggle is on `b ? null : 2`, and not simply on `null`, but I think that's OK.
